spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/InternalRowWrapper.java

```diff
@@ -40,9 +43,12 @@ class InternalRowWrapper implements StructLike {
   private InternalRow row = null;
 
   @SuppressWarnings("unchecked")
-  InternalRowWrapper(StructType rowType) {
+  InternalRowWrapper(StructType rowType, Types.StructType icebergStruct) {
```

Comment:
should we have a Pre-condition check that icebergStruct.fields().length is equal to types.lenght ?

Reply:
I've added one